tests/test_error.rs

```diff
@@ -49,9 +63,74 @@ enum EnumError {
     Unit,
 }
 
+#[derive(Error, Debug)]
+#[error(bound = std::fmt::Display + std::error::Error + 'static)]
+enum WithGeneric<T> {
+    Variant,
+    Generic(T),
+}
+
+#[derive(Error, Debug)]
+#[error(bound = std::fmt::Debug + std::error::Error + 'static)]
+enum WithGenericFrom<T> {
+    Variant,
+    Generic(#[from] T),
+}
+
+#[derive(Error, Debug)]
+#[error(bound = std::fmt::Display + std::fmt::Debug + std::error::Error + 'static)]
+enum WithGenericTransparent<T> {
+    #[error("variant")]
+    Variant,
+    #[error(transparent)]
+    Generic(#[from] T),
+}
+
+#[derive(Error, Debug)]
+#[error(bound = std::error::Error + 'static)]
+struct WithGenericStruct<T> {
+    #[from]
+    inner: T,
+}
+
+#[derive(Error, Debug)]
+#[error(bound = std::error::Error + 'static)]
+struct WithGenericStructRef<'a, T> {
+    inner: &'a WithGenericStruct<T>,
```

Comment:
Is `WithGenericStructRef` based on a real world use case? Or do your real world uses pretty much all look like one of the ones above, where the type parameter is used as the whole field type?

Reply:
I have at least one of this cases occurring in my current codebase.

I am, however, seeing some difficulties wherein I'd like to be able to "skip" bounds on generics that only exist due to a lack of universal type qualifiers on content fields in enums. That, or some way of specifying bounds on a per-generic-parameter basis.

Reply:
Unless you have a suggested solution here, I've fixed the other issue and believe that, while inconvenient, this change does help quite a bit in my code. It may be a stepping stone toward a more robust solution?